spmd/tensor/device_mesh.py

```diff
@@ -481,10 +481,30 @@ def reduce_scatter(
         num_chunks = self.size(mesh_dim)
         if self._backend == "nccl":
             input_list = list(input.tensor_split(num_chunks, dim=tensor_dim))
-            to_scatter = [tensor.contiguous() for tensor in input_list]
-            output = torch.empty_like(input_list[my_coordinate])
+            # gloo does not support uneven scattering, nccl supports it, but it
+            # might be slow compare to even size collective, we need to pad tensor
+            # before really calling scatter, and unpad/narrow it after the collective
+            # TODO: consider if we should remove this logic once ProcessGroupGloo
+            # suupport uneven list, and collective perfomance on par
+            idx_start_to_pad = input.size(tensor_dim) % num_chunks
+            to_scatter = []
+            for i, scatter_tensor in enumerate(input_list):
+                if idx_start_to_pad != 0 and i >= idx_start_to_pad:
+                    scatter_tensor = self._pad_tensor_dim_by_1(
+                        scatter_tensor, tensor_dim
+                    )
+                scatter_tensor = scatter_tensor.contiguous()
+                to_scatter.append(CommTensor(scatter_tensor))
```

Comment:
Do we need to use CommTensor explicitly here?

Reply:
Yeah we need to use CommTensor explicitly there so that CommTensor could inline the fx Node to the graph.